script/dist-info.ts

```diff
@@ -160,6 +160,11 @@ export function getDistArchitecture(): 'arm64' | 'x64' | 'armv7l' {
     return 'armv7l'
   }
 
+  const targetArch = process.env.TARGET_ARCH
+  if (targetArch === 'arm64') {
+    return 'arm64'
+  }
```

Comment:
you shouldn't need to do this. see the above lines for setting `npm_config_arch` the function should already return with that having been set